octoprint_mrbeam/software_update_information.py

```diff
@@ -19,72 +24,197 @@
+                serverfile_verson = serverfile_info["version"]
+            except ValueError as e:
+                servererror = True
+                _logger.error(
+                    "there is a wrong configured config file on the server - cancel loading of new file"
+                )
+                _logger.debug("error %s - %s", e, serverfile_request.content)
+
+        #check if local file exists valid
+        if not os.path.exists(
+                get_sw_update_file_path(self._plugin)
+        ):
+            localfilemissing = True
+
+        #if both valid compare and override
+        if not servererror and serverfile_request:
+            if localfilemissing:   # create new file
+                readwriteoption = "w+"
+                newfile = True
+            else: # checks if the file is available otherwise it will be created
+                readwriteoption = "r+"
+            with open(
+                    get_sw_update_file_path(self._plugin),
+                    readwriteoption,
+            ) as f:
+                try:
+                    update_info = json.load(f)
+                    _logger.debug("serverfile_info %s", serverfile_info)
+                    _logger.debug(
+                        "version compare %s - %s",
+                        serverfile_verson,
+                        update_info["version"],
+                    )
+                    if serverfile_verson > update_info["version"]:
+                        newfile = True
+                        _logger.info(
+                            "update local file from server - %s -> %s",
+                            serverfile_verson,
+                            update_info["version"],
+                        )
+
+                except ValueError:
+                    newfile = True
+                    _logger.error(
+                        "there is a wrong configured local config file - override local file with server file"
+                    )
+                except KeyError:
+                    localfilemissing = True
+                    _logger.exception("there is a keyerror in the local file")
+
+                #if local config file invalid override by server
+                if newfile or localfilemissing:
+                    _logger.debug("override local file")
+                    # override local file
+                    f.seek(0)
+                    f.write(serverfile_request.content)
+                    f.truncate()
+                    returncode = self.CLOUD_FILE
+
+        # if server invalid use local file
+        if not serverfile_request:
+            servererror = True
+            returncode = self.LOCAL_FILE
+
+        # if both invalid use repo file
+        if servererror and localfilemissing:
+            _logger.info(
+                "fallback use local default config file /files/software_update/update_info.json"
+            )
+            try:
+                copy(
```

Comment:
please write `shutil.copy` as it is less confusing (`copy()` is usually an object copy, from the `copy` package.)

Reply:
ok